### DISCUSSION TITLE
Custom Offline Dependencies: Allow Default Registry URL with `file://` Prefix Accessible Anywhere

### DISCUSSION THREAD

**Discussion by danhlee329** [renovatebot/renovate#27675] [Suggest an Idea]
### Tell us more.

Following up from discussion https://github.com/renovatebot/renovate/discussions/27644#discussion-6293523, would like to request that the `defaultRegistryUrlTemplate` property with the `file://` prefix (in it's value) be accessible anywhere as opposed to only from the cloned Github repo.  There may be cases where the file could be provided outside of the target Github repo we want to apply Renovate to.

**Comment by rarkins** [comment]
@viceice if the bot admin allows it, why not?

**Reply by viceice** [reply]
We can probably have an explicit paths whitelist (glob/regex), which needs to be configured by bot admin.
Shoulds be disabled (empty) by default.

**Reply by viceice** [reply]
path to repo is always allowed

**Reply by danhlee329** [reply]
forgot to mention in the discussion above that this could be done via configuration per previous discussion, whitelisting paths would be no issue for our use-case

**Reply by viceice** [reply]
the config option should have a big warning about security implications.

**Reply by danhlee329** [reply]
I feel that having a big warning about security implications will not make this a friendly nor inviting feature for users to consume.  not saying that we shouldn't have a warning, just wondering if there is a way we can do this with better inherent security

**Reply by danhlee329** [reply]
I should note that this is not a blocker for us when we run Renovate as we're hosting a local nginx service that hosts our version file, which is accessible via the URL to the file in the property `defaultRegistryUrlTemplate`.  I'd say that this request is not a biggie as long as we update the documentation stating that by using `file://`, it assumes that the file is local in the current GH repo.
